addons/account/views/account_journal_views.xml

```diff
@@ -56,38 +55,29 @@
                                 <group>
                                     <group string="Accounting Information">
                                         <field name="default_account_type" invisible="1"/>
-                                        <!-- Bank -->
-                                        <field name="default_account_id" string="Bank Account"
-                                               attrs="{'required': [('id', '!=', False), ('type', '=', 'bank')], 'invisible': [('type', '!=', 'bank')]}"
+                                        <label for="default_account_id" string="Bank Account"
+                                               attrs="{'invisible': [('type', '!=', 'bank')]}" groups="account.group_account_readonly"/>
+                                        <label for="default_account_id" string="Cash Account"
+                                               attrs="{'invisible': [('type', '!=', 'cash')]}" groups="account.group_account_readonly"/>
+                                        <label for="default_account_id" string="Default Income Account"
+                                               attrs="{'invisible': [('type', '!=', 'sale')]}" groups="account.group_account_readonly"/>
+                                        <label for="default_account_id" string="Default Expense Account"
+                                               attrs="{'invisible': [('type', '!=', 'purchase')]}" groups="account.group_account_readonly"/>
+                                        <label for="default_account_id" string="Default Account"
+                                               help="If set, this account is used to automatically balance entries."
+                                               attrs="{'invisible': [('type', '!=', 'general')]}" groups="account.group_account_readonly"/>
+                                        <field name="default_account_id" nolabel="1"
+                                               attrs="{'required': ['|', '&amp;', ('id', '!=', False), ('type', 'in', ('bank', 'cash')), ('type', 'in', ('sale', 'purchase'))], }"
                                                options="{'no_quick_create': True}"
                                                groups="account.group_account_readonly"/>
```

Comment:
You need the invisible here in case no journal type is set yet, to avoid having the field without label breaking the view
```suggestion
                                        <field name="default_account_id" nolabel="1"
                                               attrs="{'required': ['|', '&amp;', ('id', '!=', False), ('type', 'in', ('bank', 'cash')), ('type', 'in', ('sale', 'purchase'))], 'invisible': [('type', '=', False)]}"
                                               options="{'no_quick_create': True}"
                                               groups="account.group_account_readonly"/>
```